File: backend/src/crowntest/utils/FileUtils.ts
import fs from 'fs';
import path from 'path';

export class FileUtils {
  private constructor() {
    // utility class
  }

  public static ensureDirectoryExists(directory: string): void {
    const normalized = path.resolve(directory);
    if (!fs.existsSync(normalized)) {
      fs.mkdirSync(normalized, { recursive: true });
    }
  }
}

File: crowntest-sdk/src/main/java/com/crowntest/sdk/utils/FileUtils.java
package com.crowntest.sdk.utils;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public final class FileUtils {

    private FileUtils() {
        // Utility class
    }

    public static void ensureDirectoryExists(Path directory) {
        try {
            if (!Files.exists(directory)) {
                Files.createDirectories(directory);
            }
        } catch (IOException exception) {
            throw new RuntimeException("Unable to create directory: " + directory, exception);
        }
    }
}
